src/nupic/engine/Link.cpp

```diff
@@ -476,7 +474,7 @@ void Link::serialize(std::ostream &f) {
     Array a = dest_->getData().subset(destOffset_, srcCount);
     f << a; // our part of the current Dest Input buffer.
 
-    std::deque<Array>::iterator itr;
+    boost::circular_buffer<Array>::iterator itr;
```

Comment:
why this change from std:: ?

Reply:
I started to make that a std::deque but discovered that it breaks other code that should be in a different PR.  It's part of a PR to remove dimensions which ends up removing LinkPolicy and removes a lot of complexity in the Link object.  So lets delay that for now.